rmw_fastrtps_cpp/src/subscription.cpp

```diff
@@ -225,8 +232,30 @@ create_subscription(
     return nullptr;
   }
 
+  info->dds_participant_ = dds_participant;
+  info->subscriber_ = subscriber;
+  info->topic_name_mangled_ = topic_name_mangled;
+  info->topic_ = topic.desc;
   des_topic = topic.desc;
 
+  // Create ContentFilteredTopic
+  if (subscription_options->content_filter_options) {
+    rmw_subscription_content_filter_options_t * options =
+      subscription_options->content_filter_options;
+    if (nullptr != options->filter_expression) {
+      eprosima::fastdds::dds::ContentFilteredTopic * filtered_topic = nullptr;
+      if (!rmw_fastrtps_shared_cpp::create_content_filtered_topic(
+          dds_participant, des_topic,
```

Comment:
is in all DDS implementations possible to create a "content filtered topic" from a topic?

The proposed API forces all implementations that want to support content filtered topics to support them in any subscription.
Maybe a more limited API, with a different type for content filtered subscriptions would make things easier to other DDS implementations or nonDDS rmw implementations.

Reply:
This is the interface required by the DDS standard. The user calls `create_contentfilteredtopic` on the participant, which is specified as:

```
ContentFilteredTopic create_contentfilteredtopic(
 in string name,
 in Topic related_topic,
 in string filter_expression,
 in StringSeq expression_parameters);
```

```
2.2.2.2.1.7 create_contentfilteredtopic
This operation creates a ContentFilteredTopic. As described in 2.2.2.3, Topic-Definition Module, a ContentFilteredTopic can 
be used to do content-based subscriptions. 
The related Topic being subscribed to is specified by means of the related_topic parameter. The ContentFilteredTopic only 
relates to samples published under that Topic, filtered according to their content. The filtering is done by means of evaluating a 
logical expression that involves the values of some of the data-fields in the sample. The logical expression is derived from the 
filter_expression and expression_parameters arguments. 
The syntax of the filter expression and parameters is described in Annex B.
In case of failure, the operation will return a ‘nil’ value (as specified by the platform)
```

Reply:
> This is the interface required by the DDS standard. The user calls create_contentfilteredtopic on the participant, which is specified as

Nice!
If upgrading a `DataReader` to a `ContentFilteredDataReader` is required by the DDS standard this is less of a concern, but it might still limit future non-DDS based rmw implementations.